### ISSUE TITLE
Should iCloud files that are not yet available locally be downloaded?

### ISSUE THREAD

**Issue by Minnozz** [onivim/oni2#3010]
When an iCloud folder is opened in the Explorer, files that are not available locally are displayed as follows:
```
.foo.txt.icloud
```
Should they be displayed as `foo.txt` instead and downloaded from iCloud when they are opened?

**Comment by bryphe** [comment]
Indeed, this isn't being handled correctly in the moment - we should show the proper name and download on demand

This resource might be helpful for us in implementing: https://apple.stackexchange.com/questions/328329/is-there-a-way-via-the-command-line-to-cause-icloud-files-to-download

Related to #2707
